types/imagemin/index.d.ts

```diff
@@ -1,23 +1,26 @@
-// Type definitions for imagemin 6.0
+// Type definitions for imagemin 7.0
 // Project: https://github.com/imagemin/imagemin#readme
 // Definitions by: Romain Faust <https://github.com/romain-faust>
+//                 Jeff Chan <https://github.com/hkjeffchan>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
-declare function imagemin(input: ReadonlyArray<string>, outputOrOptions?: string | imagemin.Options): Promise<imagemin.Result[]>;
-declare function imagemin(input: ReadonlyArray<string>, output?: string, options?: imagemin.Options): Promise<imagemin.Result[]>;
+declare function imagemin(input: string[], options?: imagemin.Options): Promise<imagemin.Result[]>;
 
 declare namespace imagemin {
     type Plugin = (input: Buffer) => Promise<Buffer>;
 
     interface Options {
```

Comment:
You should create a separate type for `buffer`'s options. In [the documentation](https://github.com/imagemin/imagemin#options-1) the buffer options does not accept the `destination` and `glob` properties

Reply:
Sorry, I did so in PR #38193 and it seems it fails some testing.

Reply:
Updated with BufferOption